packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,132 @@ class GridSelectionColumn extends GridColumn {
+      if (this.__lassoCurrentY < scrollableArea.top) {
+        hoveredIndex = this._grid._firstVisibleIndex;
+      } else if (this.__lassoCurrentY > scrollableArea.bottom) {
+        hoveredIndex = this._grid._lastVisibleIndex;
+      }
+
+      if (hoveredIndex !== undefined) {
+        // Select all items between the start and the current row
+        renderedRows.forEach((row) => {
+          if (
+            (hoveredIndex > this.__lassoDragStartIndex &&
+              row.index >= this.__lassoDragStartIndex &&
+              row.index <= hoveredIndex) ||
+            (hoveredIndex < this.__lassoDragStartIndex &&
+              row.index <= this.__lassoDragStartIndex &&
+              row.index >= hoveredIndex)
+          ) {
+            if (this.__lassoSelect) {
+              this._grid.selectItem(row._item);
+            } else {
+              this._grid.deselectItem(row._item);
+            }
+          }
+        });
+      }
+
+      // Start scrolling in the top/bottom 15% of the scrollable area
+      const scrollTriggerArea = scrollableArea.height * 0.15;
+      // Maximum number of pixels to scroll per iteration
+      const maxScrollAmount = 10;
+
+      if (this.__lassoDy < 0 && this.__lassoCurrentY < scrollableArea.top + scrollTriggerArea) {
+        const dy = scrollableArea.top + scrollTriggerArea - this.__lassoCurrentY;
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop -= percentage * maxScrollAmount;
+      }
+      if (this.__lassoDy > 0 && this.__lassoCurrentY > scrollableArea.bottom - scrollTriggerArea) {
+        const dy = this.__lassoCurrentY - (scrollableArea.bottom - scrollTriggerArea);
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop += percentage * maxScrollAmount;
+      }
+
+      // Schedule the next auto scroll
+      setTimeout(() => this.__lassoAutoScroller(), 10);
+    }
+  }
+
+  /**
+   * Gets the scrollable area of the grid as a bounding client rect. The
+   * scrollable area is the bounding rect of the grid minus the header and
+   * footer.
+   *
+   * @private
+   */
+  __getScrollableArea() {
+    const gridRect = this._grid.$.table.getBoundingClientRect();
+    const headerRect = this._grid.$.header.getBoundingClientRect();
+    const footerRect = this._grid.$.footer.getBoundingClientRect();
+
+    return {
+      top: gridRect.top + headerRect.height,
+      bottom: gridRect.bottom - footerRect.height,
+      left: gridRect.left,
+      right: gridRect.right,
+      height: gridRect.height - headerRect.height - footerRect.height,
+      width: gridRect.width,
+    };
+  }
+
+  /** @private */
+  __onSelectionColumnCellTrack(event) {
+    if (!this.selectRowsByDragging) {
+      return;
+    }
+    this.__lassoCurrentY = event.detail.y;
+    this.__lassoDy = event.detail.dy;
+    if (event.detail.state === 'start') {
+      this._grid.setAttribute('disable-text-selection', true);
```

Comment:
Disabling / enabling text selection is missing a test.

Reply:
Test added